src/Protocol/ActivityPub/Receiver.php

```diff
@@ -1257,17 +1287,23 @@ private static function processAttachmentUrls(array $object, array $object_data)
 				continue;
 			}
 
-			if ($mediatype == 'text/html') {
-				$object_data['alternate-url'] = $href;
-			}
-
 			$filetype = strtolower(substr($mediatype, 0, strpos($mediatype, '/')));
 
 			if ($filetype == 'audio') {
 				$attachments[$filetype] = ['type' => $mediatype, 'url' => $href, 'height' => null, 'size' => null];
 			} elseif ($filetype == 'video') {
 				$height = (int)JsonLD::fetchElement($url, 'as:height', '@value');
+				// PeerTube audio-only track
+				if ($height === 0) {
+					continue;
+				}
+
 				$size = (int)JsonLD::fetchElement($url, 'pt:size', '@value');
+				// For embedded video we take the smallest available size
+				if (!empty($attachments[$mediatype]['size']) && ($size > $attachments[$mediatype]['size'])) {
+					continue;
```

Comment:
The attachment system is build in a way that all resolutions are stored and later we decide which resolution to display.

Reply:
This wasn't the case in this method where only one attachment was saved for each media type because of the array key used. Instead of relying on the order of the Link in the AP message, we now have consistent policies both for the video and the torrent links.

Reply:
Seems that at some point I introduced a bug. The plan is to store all resolutions and later decide which to display. For this reason there is this code here: https://github.com/friendica/friendica/blob/develop/src/Model/Post/Media.php#L593-L602

Reply:
Got it, I can make it more versatile.